mono/metadata/handle.h

```diff
@@ -312,23 +363,40 @@ Handle macros/functions
  * For example, TYPED_HANDLE_DECL(MonoObject) (see below) expands to:
  *
  * typedef struct {
+#if MONO_TYPE_SAFE_HANDLES
+ *   MonoObject **__raw;
+#else
  *   MonoObject *__raw;
+#endif
  * } MonoObjectHandlePayload;
  *
  * typedef MonoObjectHandlePayload* MonoObjectHandle;
  * typedef MonoObjectHandlePayload* MonoObjectHandleOut;
  */
+
+#if MONO_TYPE_SAFE_HANDLES
+#define TYPED_HANDLE_DECL(TYPE)						\
+	typedef struct { TYPE **__raw; } TYPED_HANDLE_PAYLOAD_NAME (TYPE) ; \
+	typedef TYPED_HANDLE_PAYLOAD_NAME (TYPE) TYPED_HANDLE_NAME (TYPE), \
+					         TYPED_OUT_HANDLE_NAME (TYPE)
+#else
 #define TYPED_HANDLE_DECL(TYPE)						\
 	typedef struct { TYPE *__raw; } TYPED_HANDLE_PAYLOAD_NAME (TYPE) ; \
 	typedef TYPED_HANDLE_PAYLOAD_NAME (TYPE) * TYPED_HANDLE_NAME (TYPE); \
 	typedef TYPED_HANDLE_PAYLOAD_NAME (TYPE) * TYPED_OUT_HANDLE_NAME (TYPE)
+#endif
+
 /*
  * TYPED_VALUE_HANDLE_DECL(SomeType):
  *   Expands to a decl for handles to SomeType (which is a managed valuetype (likely a struct) of some sort) and to an internal payload struct.
  * For example TYPED_HANDLE_DECL(MonoMethodInfo) expands to:
  *
  * typedef struct {
+#if MONO_TYPE_SAFE_HANDLES
```

Comment:
Fix comment, add ` * ` at beginning of the line.

Reply:
Sure.